modules/lfs/server.go

```diff
@@ -568,11 +566,39 @@ func parseToken(authorization string) (*models.User, *models.Repository, string,
 		user, password := cs[:i], cs[i+1:]
 		u, err := models.GetUserByName(user)
 		if err != nil {
-			return nil, nil, "basic", err
-		}
-		if !u.ValidatePassword(password) {
-			return nil, nil, "basic", fmt.Errorf("Basic auth failed")
+			if models.IsErrUserNotExist(err) {
+				isUsernameToken := len(password) == 0 || password == "x-oauth-basic"
+				authToken := user
+
+				if !isUsernameToken {
+					authToken = password
+				}
+
+				token, err := models.GetAccessTokenBySHA(authToken)
+				if err != nil {
+					return nil, nil, "basic", fmt.Errorf("Token not found")
+				}
+
+				u, err = models.GetUserByID(token.UID)
+				if err != nil {
+					return nil, nil, "basic", err
+				}
+
+				token.UpdatedUnix = util.TimeStampNow()
```

Comment:
Why update the access token time?

Reply:
I based it off the implementation in [routers/repo/http.go](https://github.com/go-gitea/gitea/blob/fe78154895d8809c157ed583992edf5d6f66d5dd/routers/repo/http.go#L29) and diden't realize it would be redundant. 
It should be fixed shortly.